### ISSUE TITLE
Site editor: Revisit the appearance of of hover outlines for blocks

### ISSUE THREAD

**Issue by jasmussen** [WordPress/gutenberg#44776]
When editing in the site editor, there are block boundary outlines shown on hover, in addition to select:

![before](https://user-images.githubusercontent.com/1204802/194537795-a390ab5f-9711-4c1e-9fe1-8571f6ba1098.gif)

Notably the hover outlines can feel a bit noisy and distracting at times. Should we remove them? [This draft PR](#44774) explores doing so:

![no outlines on hover](https://user-images.githubusercontent.com/1204802/194537569-d13ad0ca-748d-44fd-8e9f-d36cb1038c7c.gif)

An alternate approach is to reduce the emphasis of the outlines, make them less distracting [This draft PR](44775) explores just that, with thin text-color outlines:

![thin text color outlines on hover](https://user-images.githubusercontent.com/1204802/194538973-03ecc424-ac68-46cb-b454-8796fb2c2be3.gif)

Try the PRs out, get a feel for it. What's a good configuration of these? Keep in mind, we can adjust:

* Opacity/color
* Thickness
* Animation

Note, for color it's probably best to lean towards text-color or theme color, as arbitrary colors can blend into backgrounds otherwise.

**Comment by javierarce** [comment]
Since the outlines will still appear in the Select mode, I would vote for [the option that removes the borders](https://github.com/WordPress/gutenberg/pull/44774). It would be great to have a keyboard shortcut to switch back and forth between modes easily.

**Comment by richtabor** [comment]
Agreed. I find them noisy/distracting for sure. 

What’s the purpose of the different interactions here, between the site and block editors? Because of the header/footer parts?

**Comment by jameskoster** [comment]
Personally I find the hover outlines to be very useful when working with the more complex layouts we tend to find in templates. Consider the gif below:

![outlines](https://user-images.githubusercontent.com/846565/197216450-6fcf04f4-191e-4658-b776-60c373c063c6.gif)

Here the outlines are critical in helping me determine which block will be selected on click.

Though I would say that they (the outlines) do not need to persist on selection.

**Comment by jasmussen** [comment]
Any preference on the attached PRs? Or do we need to try a different direction?

**Comment by jameskoster** [comment]
For me the noise is mostly an issue when you've already selected a block:

![outlines](https://user-images.githubusercontent.com/846565/197521735-a772a4b8-7e5a-48ee-ad9b-9d7a42401029.gif)

Maybe we could try a version that removes the selected block outline? 🤔

**Comment by jasmussen** [comment]
Oh that's interesting. That could be worth a shot.

I also think we're at a point where we can kill the floating inserter button, don't you think?

**Comment by jameskoster** [comment]
The black one? Potentially, yes!

**Comment by jasmussen** [comment]
Rebased both PRs and recorded fresh GIFs in a gray theme that may be more representative. Just to compare, here's trunk:

![trunk](https://user-images.githubusercontent.com/1204802/200331132-a70cfa0f-dc75-43ad-8977-546f516fec0f.gif)

**No outlines:**
![no outlines](https://user-images.githubusercontent.com/1204802/200329978-3c44d7f8-5e3c-4ca9-aff5-0ffa6a7e9e7b.gif)

PR #44774 

* There are still outlines when you select a block (1.5px)
* There are still outlines on hovering template parts

**Subtle outlines:**

![subtle outlines](https://user-images.githubusercontent.com/1204802/200331546-ac7b38c0-09d7-44a6-850f-15538497eaf1.gif)

PR #44775

* 0.5px hover outlines
* Coming back to this one, I don't think it's working that well. Feels like either the outlines should be visible, or not at all.

**Inverted outlines:**

![inverted](https://user-images.githubusercontent.com/1204802/200333908-408598ce-32a7-4fea-8ad8-1a2dbbec2800.gif)

Just a quick experiment, didn't work well at all since the selected style overlaps with focused style. But nevertheless showing for comparison: 1.5px hover, 1px selected.

**No selected outline:**

Just to try what we discussed, no outline for the selected state:

![no selected](https://user-images.githubusercontent.com/1204802/200339495-0f7e62cb-ab0c-4836-9632-dbfb8078f257.gif)

It doesn't really work that well in practice, because there's still an outline on block focus. So it just means the outline disappears when selecting outside.

So at this point, I think we might need to do curate and refine what we have, and perhaps revisit at a later time. For example, an easier way to deselect all blocks (click header, etc.), changes to select mode, and further refinements to when/how often the sibling inserter appears. What do you think?

**Comment by jasmussen** [comment]
Created a followup issue and PR to improve things a bit: https://github.com/WordPress/gutenberg/issues/45590 and https://github.com/WordPress/gutenberg/pull/45589

**Comment by jameskoster** [comment]
> and perhaps revisit at a later time

Yeah I agree. It feels like we should take a comprehensive look at everything that selecting a block entails. With the site editor there are so many nuances to consider... getting it right will require a deeper dive.

**Comment by jameskoster** [comment]
Just leaving a note here for when we revisit.

Another argument for the borders is that they help to clarify the space a block occupies within a layout:

<img src="https://user-images.githubusercontent.com/5323259/200529033-f6d2f98b-05aa-460f-b31d-7301e3758f32.jpg">

This will be relevant when we get to merging https://github.com/WordPress/gutenberg/pull/45364 and other advanced layout tools.

**Comment by nekohayo** [comment]
If anything, I would want _more_ hover borders for blocks (especially container-type blocks) everywhere in the Gutenberg editor (when editing a page or post for example), not less; as it is currently, it is very difficult to be able to "spot" invisible interactive elements (including spacers, columns, groups, rows, stacks, etc.). Is there a ticket specifically about that problem?

**Comment by richtabor** [comment]
> For me the noise is mostly an issue when you've already selected a block

Agree here, though you're almost always with a selected block in the site editor.  

This here is quite distracting: 

https://github.com/WordPress/gutenberg/assets/1813435/b2c2ff33-07f1-491a-825a-f3aeaf135a39

**Comment by jameskoster** [comment]
I think that is more related to the menu and could be resolved when the block tools menu switches to the new component.

First menu here is the current component, second is the new one:

https://github.com/WordPress/gutenberg/assets/846565/fc6e4a99-833a-4302-a29b-02656a68c062
